chef/cookbooks/network/recipes/default.rb

```diff
@@ -90,7 +90,8 @@
   subscribes :run, resources(cookbook_file: "modprobe-bridge.conf"), :delayed
 end
 
-provisioner = search(:node, "roles:provisioner-server")[0]
+provisioner_instance = node[:provisioner][:config][:environment].gsub(/^provisioner-config-/, "") rescue "default"
+provisioner = node_search_with_cache("roles:provisioner-server", provisioner_instance).first
```

Comment:
This is the second time I see these two lines - maybe it's worth extracting to the utils module?